spec/faraday/connection_spec.rb

```diff
@@ -639,6 +639,24 @@ def decode(params)
 
       it_behaves_like 'default connection options'
     end
+
+    context 'preserving a user_agent assigned via default_conncetion_options' do
+      context 'when url is a Hash' do
+        let(:conn) { Faraday.new(url: 'http://example.co', headers: { 'CustomHeader' => 'CustomValue' }) }
+
+        before { Faraday.default_connection_options = { headers: { user_agent: 'My Agent 1.2' } } }
```

Comment:
I just realised there no rollback of the original value after these test have run 😄.
I'll be fixing this on the main branch, so no worries at all, but please keep this in mind next time you write tests for a global configuration 👍